server/lib/nl/detection/place.py

```diff
@@ -302,6 +305,9 @@ def _get_stripped_string(query_parts: List[QueryPart],
       # if any of the dcids in the query part is resolved, consider the query part
       # resolved and add the original string to resolved_things_str
       resolved_dcids_str.append(p.substr)
+      if len(stripped_query_parts) > 0:
+        stripped_query_parts[-1] = re.sub(_PLACE_PREPOSITION_REGEX, "",
+                                          stripped_query_parts[-1])
```

Comment:
Why are doing any stop-word related things here?  This just strips out resolved place names?

Reply:
Now I try to preserve "in", but only remove if it s before a place like this case. Concretely:

For query like "people in the USA", here removes "USA" and left with "people in the" and I want to remove "in the";

Reply:
Or should we have another pass on the queryset list and process the stop words there?

Reply:
It is very subtle and feels error prone!  What do you think of introducing a new string altogether with place names replaced with "$_PLACE" or something.  Then downstream in the stop-words replacement, add "in the $_PLACE" as replacement string.